File: rb/features.rb
# ZefoyTiktokBot SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ZefoyTiktokBotFeatures
  def self.make_feature(name)
    case name
    when "base"
      ZefoyTiktokBotBaseFeature.new
    when "test"
      ZefoyTiktokBotTestFeature.new
    else
      ZefoyTiktokBotBaseFeature.new
    end
  end
end

File: py/features.py
# ZefoyTiktokBot SDK feature factory

from feature.base_feature import ZefoyTiktokBotBaseFeature
from feature.test_feature import ZefoyTiktokBotTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ZefoyTiktokBotBaseFeature(),
        "test": lambda: ZefoyTiktokBotTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
